scripts/airgapped/retag-images-to-cache.py

```diff
@@ -0,0 +1,87 @@
+import argparse
+import logging
+
+import docker
+
+from utils import (delete_file_if_exists, get_images_list_from_file,
+                   get_or_pull_image)
+
+cli = docker.client.from_env()
+
+log = logging.getLogger(__name__)
+
+RETAGGED_IMAGES_FILE = "retagged-images.txt"
+SHA_TOKEN = "@sha256"
+REGISTRIES = ["docker.io/library", "docker.io", "gcr.io", "quay.io",
+              "registry.k8s.io", "k8s.gcr.io", "public.ecr.aws", "ghcr.io",
+              "nvcr.io"]
+
+
+def retag_image_with_sha(image):
+    """Retag the image by using the sha value."""
+    log.info("Retagging image digest: %s", image)
+    repo_digest = image.attrs["RepoDigests"][0]
+    [repository_name, sha_value] = repo_digest.split("@sha256:")
+
+    tagged_image = "%s:%s" % (repository_name, sha_value)
+    log.info("Retagging to: %s", tagged_image)
+    image.tag(tagged_image)
+
+    log.info("Tagged image successfully: %s", tagged_image)
+    return cli.images.get(tagged_image)
+
+
+def get_retagged_image_name(image_nm: str, new_registry: str) -> str:
+    """Given an image name replace the repo and use sha as tag."""
+    if SHA_TOKEN in image_nm:
+        log.info("Provided image has sha. Using it's value as tag.")
+        image_nm = image_nm.replace(SHA_TOKEN, "")
+
+    if len(image_nm.split("/")) == 1:
+        # docker.io/library image, i.e. ubuntu:22.04
+        return "%s/%s" % (new_registry, image_nm)
+
+    if len(image_nm.split("/")) == 2:
+        # classic docker.io image, i.e. argoproj/workflow-controller
+        return "%s/%s" % (new_registry, image_nm)
+
+    for registry in REGISTRIES:
+        if registry not in image_nm:
+            continue
+
+        return image_nm.replace(registry, new_registry)
```

Comment:
The above logic was not correct as it would only use one level from the right during split, but we actually need the opposite to only replace the registry.

Note that we still need the two cases above to catch the docker.io registry. So we end up with the following 3 cases:
1. `ubuntu:22.04` (which is `docker.io/ubuntu:22.04`)
2. `argoproj/workflow-controller` (which is `docker.io/argoproj/workflow-controller`)
3. `quay.io/metallb/speaker:v0.13.3`

Pushed a commit that properly handles these 3 cases 
https://github.com/canonical/bundle-kubeflow/pull/682/commits/3a3ae370d69a10e9f69baa926662920f2bb63301